### PR TITLE
dts: arm: stm32f4: Add quadspi node

### PR DESCRIPTION
**Pull request by KimBP**
By some reason qspi support has never been added to stm32f4xx

Think this is what is needed

### CODE REVIEW

**Comment by erwango** [comment]
@KimBP Have you been able to test this ?
Beides, would you be able to enable qspi in one F4 board upstream ?

**Comment by KimBP** [comment]
Unfortunately haven't tested yet. 
I have access to a STM32F412G_Disco board so it ought be possible. 
I'll enable and test it on that board and then get back with an update to this PR

**Comment by KimBP** [comment]
@erwango : stm32f412g_Disco board now updated. 
Changes verified with sample drivers_spi_flash:

*** Booting Zephyr OS build zephyr-v2.6.0-1515-g82d8dcd102d0  ***

JEDEC QSPI-NOR SPI flash testing
==========================

Test 1: Flash erase
Flash erase succeeded!

Test 2: Flash write
Attempting to write 4 bytes
Data read matches data written. Good!!

**Comment by erwango** [comment]
Thanks @KimBP!

File: dts/arm/st/f4/stm32f4.dtsi [hunk 1]
```diff
@@ -470,6 +470,17 @@
 			status = "disabled";
 			label = "DMA_2";
 		};
+
+		quadspi: quadspi@a0001000 {
+			compatible = "st,stm32-qspi";
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			reg = <0xa0001000 0x34>;
```
Comment (erwango) [inline dts/arm/st/f4/stm32f4.dtsi:478]: Nitpick: Length is not used today, but for now convention is to set it as 0x400 for all devices.

File: dts/arm/st/f4/stm32f4.dtsi [hunk 2]
```diff
@@ -470,6 +470,17 @@
...
+			interrupts = <92 0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB3 0x2>;
```
Comment (erwango) [inline dts/arm/st/f4/stm32f4.dtsi:480]:
```suggestion
			clocks = <&rcc STM32_CLOCK_BUS_AHB3 0x00000002>;
```

File: boards/arm/stm32f412g_disco/stm32f412g_disco.dts [hunk 3]
```diff
@@ -123,3 +123,20 @@
 &rtc {
 	status = "okay";
 };
+
+&quadspi {
```
Comment (erwango) [inline boards/arm/stm32f412g_disco/stm32f412g_disco.dts:127]: Please update board documentation accordingly

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this contribution.
Some minor changes, otherwise LGTM

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Comment by cfriedt** [comment]
@KimBP  - please address @FRASTM 's comments.

File: dts/arm/st/f4/stm32f4.dtsi [hunk 4]
```diff
@@ -470,6 +470,17 @@
 			status = "disabled";
 			label = "DMA_2";
 		};
+
+		quadspi: quadspi@a0001000 {
```
[the remark this answers is not in the document]
Comment (KimBP) [reply dts/arm/st/f4/stm32f4.dtsi:474]: OK - wasn't aware of that - but have had the thought. 
I'm currently interested in stm32f412xg but think we should put it where it covers as many devices as possible.
Does it exist in all stm32f412 variants?
stm32f412.dtsi by the way includes stm32f411 (which again includes stm32f401).
I.e putting it into. stm32f411.dtsi could be an option too ?

**Review by KimBP — COMMENTED** [review]
@FRASTM : Will do. Originally it was set to 0x34, but erwango claimed the convention is to set it to 0x400. Don't know why

**Comment by cfriedt** [comment]
@erwango, @FRASTM - is this change ready to go? I wasn't sure if all concerns were addressed, but it has both approvals atm

**Comment by KimBP** [comment]
@cfriedt I think I've followed up on all issues being raised

**Pull request merged**